### ISSUE TITLE
Enum parsing should be case insensitive

### ISSUE THREAD

**Issue by robbiemcmichael** [phadej/github#371]
I'm working on a GitHub App and noticed that some of the enums in webhook payloads have a different capitalisation to the normal webhook payloads. This is pretty amazing since I assumed it would be the same code for both, but apparently not.

Either way, I think it would be a good idea to make enum parsing case insensitive to make the library more resilient to strange API implementations like this from GitHub's end.

I'm happy to do the implementation, just wanted to get a yes/no on the idea before I do the work.

**Comment by phadej** [comment]
something like

```haskell
parseJSON = withText "EnumName" $ \t -> case T.toLower t of
  "foo" -> ...
  "bar" -> ...
```

would be fine.

I'm planning to prepare the release today, so the sooner the PR the higher chance it will get in!

**Comment by robbiemcmichael** [comment]
Sure, I'll do it now if that means it gets into the release.

**Comment by phadej** [comment]
Will be fixed by #373
